cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsRdsTerminateService.java

```diff
@@ -117,21 +117,18 @@ private void initiateCFTemplateDeletion(
             AwsCredentialView credentialView,
             String regionName
     ) {
-        DescribeStacksRequest describeStacksRequest = new DescribeStacksRequest().withStackName(cFStackName);
         AmazonCloudFormationClient cfClient = awsClient.createCloudFormationClient(credentialView, regionName);
-        cfClient.describeStacks(describeStacksRequest);
-        DeleteStackRequest deleteStackRequest = awsStackRequestHelper.createDeleteStackRequest(cFStackName);
-        cfClient.deleteStack(deleteStackRequest);
-        LOGGER.debug("CloudFormation stack deletion request sent with stack name: '{}' for stack: '{}'", cFStackName, ac.getCloudContext().getId());
-
-        Waiter<DescribeStacksRequest> stackDeleteCompleteWaiter = cfClient.waiters().stackDeleteComplete();
-        StackCancellationCheck stackCancellationCheck = new StackCancellationCheck(ac.getCloudContext().getId());
-        WaiterParameters<DescribeStacksRequest> describeStacksRequestWaiterParameters = new WaiterParameters<>(describeStacksRequest)
-                .withPollingStrategy(getBackoffCancellablePollingStrategy(stackCancellationCheck));
-        stackDeleteCompleteWaiter.run(describeStacksRequestWaiterParameters);
-    }
-
-    private void wrapExceptionIfNeeded(Exception e) throws Exception {
-
+        if (cfStackUtil.isCfStackExists(cfClient, cFStackName)) {
```

Comment:
If _isCfStackExists_ method can throw Retry.ActionFailedException, I think it should be called with retry (like in the AwsTerminateService)
`retryService.testWith2SecDelayMax15Times(() -> cfStackUtil.isCfStackExists(amazonCloudFormationClient, cFStackName));`